pkg/internalapis/cnsvolumeoperationrequest/cnsvolumeoperationrequest.go

```diff
@@ -122,20 +108,130 @@ func InitVolumeOperationRequestInterface(ctx context.Context) (VolumeOperationRe
 // If no previous information has been persisted, it returns an empty object.
 func (or *operationRequestStoreOnETCD) GetRequestDetails(ctx context.Context, name string) (*VolumeOperationRequestDetails, error) {
 	log := logger.GetLogger(ctx)
-	// TODO: Implement GetRequestDetails for operationRequestStoreOnETCD
-	err := fmt.Errorf("GetRequestDetails not implemented for operationRequestStoreOnETCD")
-	log.Error(err)
-	return nil, err
+
+	instance := &cnsvolumeoperationrequestv1alpha1.CnsVolumeOperationRequest{}
+	key := client.ObjectKey{
+		Name:      name,
+		Namespace: csiconfig.DefaultCSINamespace,
+	}
+	log.Infof("Getting CnsVolumeOperationRequest instance with name %s/%s", csiconfig.DefaultCSINamespace, name)
+	err := or.k8sclient.Get(ctx, key, instance)
+	if err != nil {
+		// If instance does not exist, return an empty object
+		if errors.IsNotFound(err) {
+			log.Infof("CnsVolumeOperationRequest instance %s/%s not found. Returning ...", csiconfig.DefaultCSINamespace, name)
+			return nil, nil
+		}
+		log.Errorf("failed to get CnsVolumeOperationRequest instance %s/%s with error: %v", csiconfig.DefaultCSINamespace, name, err)
+		return nil, err
+	}
+	log.Debugf("Found CnsVolumeOperationRequest instance %v", spew.Sdump(instance))
+
+	// Determine which operation details need to be returned. Callers only need to know about the
+	// last operation that was invoked on a volume.
+	var operationDetailsToReturn cnsvolumeoperationrequestv1alpha1.OperationDetails
+	if len(instance.Status.LatestOperationDetails) != 0 {
+		operationDetailsToReturn = instance.Status.LatestOperationDetails[len(instance.Status.LatestOperationDetails)-1]
+	} else {
+		operationDetailsToReturn = instance.Status.FirstOperationDetails
```

Comment:
Shouldn't FirstOperationDetails also be stored in LatestOperationDetails the first time? If yes, then you wouldn't need this else statement.

Reply:
Makes sense. Updated the logic!